tests/plugins/generic/test_goggle.py

```diff
@@ -0,0 +1,199 @@
+def test_plugin_generate(test_plugin: Plugin, serialize: bool) -> None:
+    X = get_airfoil_dataset()
+    test_plugin.fit(GenericDataLoader(X))
+
+    if serialize:
+        saved = save(test_plugin)
+        test_plugin = load(saved)
+
+    X_gen = test_plugin.generate()
+    assert len(X_gen) == len(X)
+    assert X_gen.shape[1] == X.shape[1]
+    assert test_plugin.schema_includes(X_gen)
+
+    X_gen = test_plugin.generate(50)
+    assert len(X_gen) == 50
+    assert test_plugin.schema_includes(X_gen)
+
+    # generate with random seed
+    X_gen1 = test_plugin.generate(50, random_state=0)
+    X_gen2 = test_plugin.generate(50, random_state=0)
+    X_gen3 = test_plugin.generate(50)
+    assert (X_gen1.numpy() == X_gen2.numpy()).all()
+    assert (X_gen1.numpy() != X_gen3.numpy()).any()
+
+
+@pytest.mark.parametrize(
+    "test_plugin", generate_fixtures(plugin_name, plugin, plugin_args)
+)
+def test_plugin_generate_constraints_goggle(test_plugin: Plugin) -> None:
+    X, y = load_iris(as_frame=True, return_X_y=True)
+    X["target"] = y
+    test_plugin.fit(GenericDataLoader(X))
+
+    constraints = Constraints(
+        rules=[
+            ("target", "eq", 1),
+        ]
+    )
+
+    X_gen = test_plugin.generate(constraints=constraints).dataframe()
+    assert len(X_gen) == len(X)
+    assert test_plugin.schema_includes(X_gen)
+    assert constraints.filter(X_gen).sum() == len(X_gen)
+    assert (X_gen["target"] == 1).all()
+
+    X_gen = test_plugin.generate(count=50, constraints=constraints).dataframe()
+    assert len(X_gen) == 50
+    assert test_plugin.schema_includes(X_gen)
+    assert constraints.filter(X_gen).sum() == len(X_gen)
+    assert list(X_gen.columns) == list(X.columns)
+
+
+def test_sample_hyperparams() -> None:
+    for i in range(100):
+        args = plugin.sample_hyperparameters()
+        assert plugin(**args) is not None
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    "compress_dataset,decoder_arch",
+    [
+        (True, "het"),
+        (False, "het"),
+        (True, "gcn"),
+        (False, "gcn"),
+        (True, "sage"),
+        (False, "sage"),
+    ],
+)
+def test_eval_performance_goggle(compress_dataset: bool, decoder_arch: str) -> None:
+    results = []
+
+    Xraw, y = load_diabetes(return_X_y=True, as_frame=True)
+    Xraw["target"] = y
+    X = GenericDataLoader(Xraw)
+
+    for retry in range(2):
+        test_plugin = plugin(
+            n_iter=5000,
+            compress_dataset=compress_dataset,
+            decoder_arch=decoder_arch,
+            random_state=retry,
+        )
+        evaluator = PerformanceEvaluatorXGB()
+
+        test_plugin.fit(X)
+        X_syn = test_plugin.generate()
+
+        results.append(evaluator.evaluate(X, X_syn)["syn_id"])
+
+    print(plugin.name(), compress_dataset, decoder_arch, results)
+    assert np.mean(results) > 0.7
+
+
+# @pytest.mark.skipif(sys.platform != "linux", reason="Linux only for faster results")
```

Comment:
If there are no plans for conditional generation, you should delete these comments.

Reply:
I am hoping to add conditional generation as a later date, but will delete in the next push as I haven't seen the research code for it yet.